src/contexts/WalletContext/services.ts

```diff
@@ -52,6 +52,53 @@ export const checkForRiskyWallet = async (walletAddress: string) => {
   }
 };
 
+// Check if the provided signature is valid for the smart contract at signer.getAddress(),
+// using the checks defined in EIP-1271.
+//
+// For simplicity we use the user's Metamask provider and not Alchemy/Infura, though we could change
+// that in future.
+export const isValidEIP1271Signature = async (
```

Comment:
Lets write unit tests for this and keep on improving test coverage of repo 🙏

Reply:
It is nothing fancy. You can mock the third party services and write scenarios on behaviour based on mock. have a look into the getAMMs.test.ts, there we mock `client.query` which is a third party library (apollo)

Reply:
For your case you can mock the parameters you pass to this function and based on the mocked values test the flow.

Reply:
Tried to get jest mocking working with my function this morning and didn't manage it. The mocking you pointed me to (in getAMMs.test.ts) is a simpler case, and testing interactions with metamask and contracts is more complicated - not least because in the Contract class the method we want to mock () only gets added at construction time.

Mocking this stuff should be doable with some more research (e.g. see ideas here: https://github.com/ethers-io/ethers.js/discussions/3095), and I agree we should add more tests, but we need to work out the right pattern and I don't think this PR, which we want to merge to unblock a user who has funds trapped in the platform, is the right place to do that research.

So I'd like to merge this without tests, and we can open a ticket (if we don't already have one) for writing tests of functions that interface with the blockchain. wdyt?

Reply:
Sure lets pair on writing tests or I can prepare something and share with you next week 🙏 If the effort now is big we can do it later as you suggest! 

Feel free to resolve this message once you have seen it 🥰